static/canvas.js

```diff
@@ -126,8 +128,27 @@ class Canvas {
         /* Event for moving the mouse. Primarily used for enlarging a box and moving it. */
         canvas.addEventListener('mousemove', function (e) {
             var mouse = myState.getMouse(e);
-            if (!myState.enlargeDirection && !myState.dragging) {
-                return;
+            var mx = mouse.x;
+            var my = mouse.y;
+            var boxes = myState.boxes;
+            var l = boxes.length;
+            var border = false;
+            var box = null;
+
+            /* Checks if mouse is hovering over a border. */
+            for (var i = l - 1; i >= 0; i--) {
+                box = boxes[i]; 
+                border = boxes[i].withinBorder(mx, my);
+                if (border) {
+                    break;
+                }
+            }
+
+
+            if (!myState.enlargeDirection && !myState.dragging && !border) {
+                document.body.style.cursor = 'default';
+            } else if (!myState.enlargeDirection && !myState.dragging && border) {
+                myState.doubleArrow(box, mx, my);
```

Comment:
Thanks, I added contains before the double arrow function like you said and that took care of the conditionals for the vertical and horizontal arrow cases, but I still needed the conditionals for the diagonal arrows. I tried to clean this up by making another function whichCorner which outputs the corner of a box regardless of how it was dragged. So double arrow now takes in a border and corner and has much cleaner conditions. Let me know if you think I could clean this up further though, thanks!